integreat_compass/cms/models/offers/offer.py

```diff
@@ -46,6 +46,16 @@ def public_version(self):
                 return version
         return None
 
+    @cached_property
+    def latest_version(self):
+        """
+        Returns the latest version of an offer.
+
+        :return: OfferVersion
+        :rtype: ~integreat_compass.cms.models.offers.offer_version.OfferVersion
+        """
+        return list(self.versions.select_related())[-1]
```

Comment:
```suggestion
        return self.versions.select_related().first()
```
Retrieving all versions (including all related models) is a bit overkill just for the latest version. It's more efficient to perform the filtering on the sql layer and just retrieve the one record we're actually interested in.
If you tried this and it didn't work for you, it is because the date was not correctly set on a newer offer version. But if the version creation is fixed with the correct dates, the `first()` will actually return the latest version.

Reply:
Does this work for you? I know it should, but just did not for me.

Reply:
Could you head over to http://localhost:8082/admin/cms/offerversion/ and inspect the `offer_version_date` fields of the offer versions? In my case, all versions had the same date. After fixing the date and give the latest version a newer date, it worked.
So when creating a new offer version, we have to not only increment the id, but also set the date to the current date...

Reply:
So this seems to be a bug introduced by #65.

Reply:
Ah, alright. Thank you for catching this!!